test/geometries_tests.cpp

```diff
@@ -1,115 +1,202 @@
+        ring.push_back(geom::Coordinate(+0, +1));
+        ring.push_back(geom::Coordinate(+1, -1));
+        ring.push_back(geom::Coordinate(-1, -1));   //needed to close the ring!
+        linearRing->setCoordinates(ring);
+
+        // set up polygon from ring
+        Polygon::Ptr polygon(new Polygon());
+        std::vector<geom::Geometry*> holes;
+        auto polygonGeometry = geom::GeometryFactory::getDefaultInstance()->createPolygon(*linearRing->geometry(), holes);  //linear Ring as DeepCopy
+        polygon->setGeometry(polygonGeometry);
 
-        auto ring = (OGRLinearRing*) OGRGeometryFactory::createGeometry(wkbLinearRing);
-        ring->addPoint(-1, -1);
-        ring->addPoint(0, 1);
-        ring->addPoint(1, -1);
-        ring->closeRings();
+        // set up the linestring
+        LineString::Ptr linestring(new LineString());
+        std::vector<geom::Coordinate> string;
+        string.push_back(geom::Coordinate(-5, 0));
+        string.push_back(geom::Coordinate(+0, 0));
+        linestring->setCoordinates(string);
+
+        // set up a multi point
+        MultiPoint::Ptr multiPoint(new MultiPoint());
+        std::vector<geom::Geometry*> points;
+        for (auto coord : ring)
+        {
+            points.push_back(geom::GeometryFactory::getDefaultInstance()->createPoint(coord));
+        }
+        multiPoint->setCollection(points);
 
-        polygon->geometry()->addRingDirectly(ring);
+        // build a geometry collection with the point, linestring and polygon
+        std::vector<geom::Geometry*> geoms;
+        geoms.push_back(point->geometry()->clone());
+        geoms.push_back(linestring->geometry()->clone());
+        geoms.push_back(linearRing->geometry()->clone());
+        geoms.push_back(polygon->geometry()->clone());
 
-        linestring->geometry()->addPoint(-5, 0);
-        linestring->geometry()->addPoint(0, 0);
+        GeometryCollection::Ptr collection(new GeometryCollection());
+        collection->setCollection(geoms);
 
-        // insert.
+        // insert 
         core.addEntity(point);
         core.addEntity(linestring);
+        core.addEntity(linearRing);
         core.addEntity(polygon);
+        core.addEntity(collection);
+        core.addEntity(multiPoint);
     }
 
     BOOST_AUTO_TEST_CASE(geometries_operations)
     {
-        ODBStorage::Ptr storage = loadStorage(dbfile);
+        ODBStorage::Ptr storage = loadStorage(db_path);
 
         std::vector<Point::Ptr> points;
         std::vector<LineString::Ptr> lstrings;
+        std::vector<LinearRing::Ptr> lrings;
         std::vector<Polygon::Ptr> polygons;
+        std::vector<GeometryCollection::Ptr> collections;
+        std::vector<MultiPoint::Ptr> multiPoints;
 
         storage->loadAll(points);
         storage->loadAll(lstrings);
+        storage->loadAll(lrings);
         storage->loadAll(polygons);
+        storage->loadAll(collections);     // it seems that there a loading issues with the collection
```

Comment:
Does the issue (whatever it was) still persist?

Reply:
No its solved for a while. Comment removed.